src/hooks/utility/view.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2023-2024 Retake, Inc.
+//
+// This file is part of ParadeDB - Postgres for Search and Analytics
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use anyhow::{bail, Result};
+use sqlparser::{dialect::PostgreSqlDialect, parser::Parser};
+use std::ops::ControlFlow;
+
+use pgrx::*;
+use sqlparser::ast::visit_relations;
+
+use crate::duckdb::connection::{execute, view_exists};
+
+use super::{get_postgres_current_schema, set_search_path_by_pg};
+
+pub fn view_query(query_string: &core::ffi::CStr) -> Result<bool> {
+    // Use the current scheme if the schema is not provided in the query.
+    let current_schema = get_postgres_current_schema();
```

Comment:
Postgres not only use one schema to search relation.  for example
```sql
create schema s1;
create schema s2;
create table s1.t1;
create table s2.t2;

set search_path to s1, s2;
select * from t1; (ref to s1.t1)

select * from t2; (ref to s2.t2)
```
If we handle this ourselves, it can be quite complex. Therefore, we should use PostgreSQL to perform symbol resolution. I still prefer to check the PostgreSQL catalog first.